### ISSUE TITLE
NFS NIC error prevents Vnet/Subnet deletion on AKS updates

### ISSUE THREAD

**Issue by manoatsas** [sassoftware/viya4-iac-azure#108]
On AKS updates, NFS NIC resource does not allow Vnet update

```
module.aks.azurerm_kubernetes_cluster.aks: Still destroying... [id=/subscriptions/XXXX--...nerService/managedClusters/mnm-tst-aks, 4m0s elapsed]
module.aks.azurerm_kubernetes_cluster.aks: Destruction complete after 4m3s
module.vnet.azurerm_virtual_network.vnet: Destroying... [id=/subscriptions/XXXX/resourceGroups/mnm-tst-rg/providers/Microsoft.Network/virtualNetworks/mnm-tst-vnet]

Error: Error deleting Virtual Network "mnm-tst-vnet" (Resource Group "mnm-tst-rg"): network.VirtualNetworksClient#Delete: Failure sending request: StatusCode=400 -- Original Error: Code="InUseSubnetCannotBeDeleted" Message="Subnet mnm-tst-misc-subnet is in use by /subscriptions/XXXX/resourceGroups/mnm-tst-rg/providers/Microsoft.Network/networkInterfaces/mnm-tst-nfs-nic/ipConfigurations/mnm-tst-nfs-ip_config and cannot be deleted. In order to delete the subnet, delete all the resources within the subnet. See aka.ms/deletesubnet." Details=[]
```

**Comment by manoatsas** [comment]
Tried to add  `depends_on = [module.vnet]` to module.nfs that still does not seem to enforce decencies between NFS VM/NIC and VNet/Subnets.

This seems to be due to an update to tags that forces VNet Subnets. I found a similar issue open in the module repo - https://github.com/Azure/terraform-azurerm-vnet/issues/41 

```
  # module.vnet.azurerm_virtual_network.vnet must be replaced
-/+ resource "azurerm_virtual_network" "vnet" {
        address_space         = [
            "192.168.0.0/16",
        ]
        dns_servers           = []
      ~ guid                  = "e40fe24a-0818-4a80-94c9-0dce5f737f96" -> (known after apply)
      ~ id                    = "/subscriptions/XXXXX/resourceGroups/mnm-tst-rg/providers/Microsoft.Network/virtualNetworks/mnm-tst-vnet" -> (known after apply)
      ~ location              = "eastus2" -> (known after apply) # forces replacement
        name                  = "mnm-tst-vnet"
        resource_group_name   = "mnm-tst-rg"
      ~ subnet                = [
          - {
              - address_prefix = "192.168.16.0/20"
              - id             = "/subscriptions/XXXXX/resourceGroups/mnm-tst-rg/providers/Microsoft.Network/virtualNetworks/mnm-tst-vnet/subnets/mnm-tst-aks-subnet"
              - name           = "mnm-tst-aks-subnet"
              - security_group = ""
            },
          - {
              - address_prefix = "192.168.2.0/24"
              - id             = "/subscriptions/XXXXX/resourceGroups/mnm-tst-rg/providers/Microsoft.Network/virtualNetworks/mnm-tst-vnet/subnets/mnm-tst-misc-subnet"
              - name           = "mnm-tst-misc-subnet"
              - security_group = ""
            },
        ] -> (known after apply)
      ~ tags                  = {
            "cost_center"   = "rnd"
            "environment"   = "dev"
          ~ "project_name"  = "IAC" -> "testdep"
            "resourceowner" = "mameen"
        }
        vm_protection_enabled = false
    }
```

**Comment by manoatsas** [comment]
Related to
- https://github.com/Azure/terraform-azurerm-compute/issues/107
- https://github.com/hashicorp/terraform/issues/25993

**Comment by manoatsas** [comment]
Resolved with PR #123
